optlib/forth.ctags

```diff
@@ -0,0 +1,23 @@
+#
+#  Copyright (c) 2023, Eric Forgeot
+#  Copyright (c) 2023, Masatake YAMATO
+#
+#  This source code is released for free distribution under the terms of the
+#  GNU General Public License version 2 or (at your opinion) any later version.
+#
+#  This module contains functions for generating tags for Forth files
+#  https://www.forth.com/starting-forth/1-forth-stacks-dictionary/
+#
+--langdef=Forth
+--map-Forth=+.fth
+--map-Forth=+.forth
+# TODO: a selector arbitrating Fortran and Forth
+# --map-Forth=+.f
+
+--kinddef-Forth=w,word,words
+--kinddef-Forth=v,variable,variables
+--kinddef-Forth=c,constant,constants
+
+--regex-Forth=/^:[[:space:]]*([^[:space:]]+)/\1/w/
+--regex-Forth=/^(variable|VARIABLE)[[:space:]]*([^[:space:]]+)/\2/v/
+--regex-Forth=/^([[:alnum:]]+)[[:space:]](constant|CONSTANT)[[:space:]]*([^[:space:]]+)/\3/c/
```

Comment:
We can do the same as the `variable`:

```
--regex-Forth=/^([[:alnum:]]+)[[:space:]]constant[[:space:]]*([^[:space:]]+)/\2/c/{exclusive}{iscase}
```

Do you know if we need the first group?

```
--regex-Forth=/^[[:alnum:]]+[[:space:]]constant[[:space:]]*([^[:space:]]+)/\1/c/{exclusive}{iscase}
```

Reply:
I haven't kept this one, it doesn't catch the constants.

The first alnum is necessary because a number is always needed to initialise a constant. All the cases can't be catched anyway, because instead of:

 ```
10 constant DUMMY1
```

it would also be possible to write


 ```
10 
constant DUMMY1
```

but it doesn't really matter